sail/src/main/java/org/apache/rya/rdftriplestore/inference/InferenceEngine.java

```diff
@@ -405,6 +382,30 @@ public void refreshGraph() throws InferenceEngineException {
     }
 
     /**
+     * Query for and collect all instances of a given type. Should only be called for types expected
+     * to have few members, such as ontology vocabulary terms, as instances will be collected in
+     * memory.
+     */
+    private Set<URI> fetchInstances(final URI type) throws QueryEvaluationException {
+        final Set<URI> instances = new HashSet<>();
+        CloseableIteration<Statement, QueryEvaluationException> iter = RyaDAOHelper.query(
+            ryaDAO, null, RDF.TYPE, type, conf);
+        try {
+            while (iter.hasNext()) {
+                final Statement st = iter.next();
+                if (st.getSubject() instanceof URI) {
+                    instances.add((URI) st.getSubject());
+                }
+            }
+        } finally {
+            if (iter != null) {
+                iter.close();
+            }
+        }
```

Comment:
This can be reduced to:
```
            ryaDaoQueryWrapper.queryAll(null, RDF.TYPE, type, new RDFHandlerBase() {
                @Override
                public void handleStatement(final Statement st) throws RDFHandlerException {
                    if (st.getSubject() instanceof URI) {
                        instances.add((URI) st.getSubject());
                    }
                }
            });
```